src/Common/MoistThermodynamics/states.jl

```diff
@@ -78,18 +78,36 @@ struct PhaseEquil{FT, PS} <: ThermodynamicState{FT}
     "temperature: computed via [`saturation_adjustment`](@ref)"
     T::FT
 end
+
+"""
+    PhaseEquil
+
+Moist thermodynamic phase, given
+ - `param_set` an `AbstractParameterSet`, see the [`MoistThermodynamics`](@ref) for more details
+ - `e_int` internal energy
+ - `ρ` density
+ - `q_tot` total specific humidity
+
+and, optionally
+
+ - `maxiter` maximum iterations for saturation adjustment
+ - `temperature_tol` estimated temperature tolerance for saturation adjustment
+ - `sat_adjust` function pointer to particular saturation adjustment function
+"""
 function PhaseEquil(
     param_set::APS,
     e_int::FT,
     ρ::FT,
     q_tot::FT,
-    maxiter::Int = 3,
-    tol::FT = FT(1e-1),
+    maxiter::Int = 5,
+    temperature_tol::FT = FT(1e-1),
     sat_adjust::Function = saturation_adjustment,
 ) where {FT <: Real}
     # TODO: Remove these safety nets, or at least add warnings
     # waiting on fix: github.com/vchuravy/GPUifyLoops.jl/issues/104
     q_tot_safe = clamp(q_tot, FT(0), FT(1))
+    _cp_d = FT(cp_d(param_set))
+    tol = ResidualTolerance(temperature_tol*_cp_d)
```

Comment:
Add an explanation here:
```suggestion
    # Convert temperature tolerance to a convergence criterion on internal energy residuals 
    tol = ResidualTolerance(temperature_tol*_cv_d)
```